### PR TITLE
Feature/force cosmo

### PR DESCRIPTION
**Pull request by cphyc**
## PR Summary

Solves issue #1311 by providing a way to force cosmological simulation detection. This is useful in cases where the conformal time of a cosmological simulation becomes positive. The simulation would then be detected as being non cosmological.=

## PR Checklist

- [x] Code passes flake8 checker
- [x] New features are documented, with docstrings and narrative docs
- [x] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

**Comment by cphyc** [comment]
I've got a weird issue with the tests. If I run the tests with 1 or 2 processes I get:
```
> nosetests yt.frontends.ramses -v --with-timer --processes=1
yt.frontends.ramses.tests.test_outputs.test_RAMSESDataset ... ok (1.0551s)
yt : [WARNING  ] 2017-07-06 11:26:37,795 Overriding code units: Use this option only if you know that the dataset doesn't define the units correctly or at all.
yt.frontends.ramses.tests.test_outputs.test_units_override ... ok (0.0336s)
yt.frontends.ramses.tests.test_outputs.test_non_cosmo_detection ... ok (0.0201s)
yt.frontends.ramses.tests.test_outputs.test_unit_non_cosmo ... ok (0.0207s)
yt.frontends.ramses.tests.test_outputs.test_cosmo_detection ... ok (0.1558s)
yt.frontends.ramses.tests.test_outputs.test_unit_cosmo ... ok (0.1423s)

----------------------------------------------------------------------
XML: /home/ccc/Documents/prog/yt/nosetests.xml
[None] 73.91% yt.frontends.ramses.tests.test_outputs.test_RAMSESDataset: 1.0548s
[None] 10.91% yt.frontends.ramses.tests.test_outputs.test_cosmo_detection: 0.1557s
[None] 9.97% yt.frontends.ramses.tests.test_outputs.test_unit_cosmo: 0.1423s
[None] 2.35% yt.frontends.ramses.tests.test_outputs.test_units_override: 0.0335s
[None] 1.45% yt.frontends.ramses.tests.test_outputs.test_unit_non_cosmo: 0.0207s
[None] 1.41% yt.frontends.ramses.tests.test_outputs.test_non_cosmo_detection: 0.0201s
----------------------------------------------------------------------
Ran 6 tests in 1.461s

OK
```
However, if I do the same with 3 or more processes:
```
> nosetests yt.frontends.ramses -v --with-timer --processes=3
yt : [WARNING  ] 2017-07-06 11:27:11,050 Overriding code units: Use this option only if you know that the dataset doesn't define the units correctly or at all.
yt.frontends.ramses.tests.test_outputs.test_units_override ... ok (0.0436s)
yt.frontends.ramses.tests.test_outputs.test_unit_non_cosmo ... ok (0.0240s)
yt.frontends.ramses.tests.test_outputs.test_cosmo_detection ... ok (0.2327s)
yt.frontends.ramses.tests.test_outputs.test_non_cosmo_detection ... FAIL
yt.frontends.ramses.tests.test_outputs.test_unit_cosmo ... ok (0.1989s)
yt.frontends.ramses.tests.test_outputs.test_RAMSESDataset ... ok (1.2821s)

======================================================================
FAIL: yt.frontends.ramses.tests.test_outputs.test_non_cosmo_detection
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ccc/.virtualenvs/astro/lib/python3.6/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/ccc/Documents/prog/yt/yt/frontends/ramses/tests/test_outputs.py", line 71, in test_non_cosmo_detection
    assert_equal(ds.cosmological_simulation, 0)
  File "/home/ccc/.virtualenvs/astro/lib/python3.6/site-packages/numpy/testing/utils.py", line 416, in assert_equal
    raise AssertionError(msg)
AssertionError:
Items are not equal:
 ACTUAL: 1
 DESIRED: 0
>>  raise AssertionError('\nItems are not equal:\n ACTUAL: 1\n DESIRED: 0')


----------------------------------------------------------------------
XML: /home/ccc/Documents/prog/yt/nosetests.xml
[None] 57.74% yt.frontends.ramses.tests.test_outputs.test_RAMSESDataset: 1.2816s
[None] 19.79% yt.frontends.ramses.tests.test_outputs.test_non_cosmo_detection: 0.4392s
[None] 10.48% yt.frontends.ramses.tests.test_outputs.test_cosmo_detection: 0.2326s
[None] 8.96% yt.frontends.ramses.tests.test_outputs.test_unit_cosmo: 0.1989s
[None] 1.95% yt.frontends.ramses.tests.test_outputs.test_units_override: 0.0433s
[None] 1.08% yt.frontends.ramses.tests.test_outputs.test_unit_non_cosmo: 0.0240s
----------------------------------------------------------------------
Ran 6 tests in 1.310s

FAILED (failures=1)
```

Any guess on that one?

File: yt/frontends/ramses/data_structures.py [hunk 1]
```diff
@@ -630,7 +635,13 @@ def read_rhs(cast):
         # This is likely not true, but it's not clear how to determine the boundary conditions
         self.periodicity = (True, True, True)
         # These conditions seem to always be true for non-cosmological datasets
```
Comment (ngoldbaum) [inline yt/frontends/ramses/data_structures.py]: Can you move this comment inside the `else` block?

File: yt/frontends/ramses/data_structures.py [hunk 2]
```diff
@@ -630,7 +635,13 @@ def read_rhs(cast):
...
-        if rheader["time"] >= 0 and rheader["H0"] == 1 and rheader["aexp"] == 1:
+        if self.force_cosmological is not None:
+            is_cosmological = self.force_cosmological
+        else:
+            is_cosmological = (rheader["time"] >= 0 and
+                               rheader["H0"] == 1 and
+                               rheader["aexp"] == 1)
```
Comment (ngoldbaum) [inline yt/frontends/ramses/data_structures.py]: There's a logic error here. Before we had if these conditions are true, then it's *not* a cosmology simulation. But here you're saying it's a cosmology simulation if these conditions are true.

I *think* the fix is just to change this to a `not` in front of the boolean expression

Comment (cphyc) [inline yt/frontends/ramses/data_structures.py]: Indeed. It also shows the tests aren't catching the mistake.

File: yt/frontends/ramses/data_structures.py (hunk 1 above)
Comment (cphyc) [inline yt/frontends/ramses/data_structures.py]: Done!

File: yt/frontends/ramses/data_structures.py (hunk 2 above)
Comment (cphyc) [inline yt/frontends/ramses/data_structures.py]: It looks like the fix is helping to pass the test with parallelism, but I still don't understand why they were passing in serial.

**Comment by cphyc** [comment]
Just did a rebase onto master!

**Pull request merged**